app/Imports/BillingDataImport.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Imports;
+
+use Illuminate\Http\UploadedFile;
+use PhpOffice\PhpSpreadsheet\Reader\Csv;
+use App\Imports\Filters\BillingDataReadFilter;
+use Carbon\Carbon;
+
+class BillingDataImport
+{
+    /**
+     * @param int $billing_id
+     * @param UploadedFile $file
+     * @return array $preparedBillingData
+     */
+    public function getBillingDataToImport($billing_id, $file): array
+    {
+        $filter = $this->getReadFilter();
+        $reader = $this->setupReaderWithFilter($filter);
+        $loadedBillingData = $this->loadBillingDataToArray($reader, $file);
+        $preparedBillingData = $this->prepareBillingData($billing_id, $loadedBillingData);
+        return $preparedBillingData;
+    }
+
+    /**
+     * @return BillingDataReadFilter $readFilter
+     */
+    private function getReadFilter(): BillingDataReadFilter
+    {
+        $columnsToRead = ['A', 'G'];
+        return new BillingDataReadFilter($columnsToRead);
+    }
+
+    /**
+     * @param BillingDataReadFilter $filter
+     * @return Csv $reader
+     */
+    private function setupReaderWithFilter($filter): Csv
+    {
+        $reader = new Csv();
```

Comment:
protip. Zamiast CSV to wstrzykuj do tej klasy IReader. Wtedy łatwo możesz rozszerzyć funkcjonalność o inne formaty plików obsługiwane przez bibliotekę. Laravel ogarnie, że wstrzykując tę klasę powinien coś do niej wstrzyknąć przy czym z interfejsem będzie miał problem - będziesz musiał zbindować interfejs do jakiejś instancji (i to będzie defaultowy typ pliku).

Reply:
+ klasa będzie bardziej testowalna

Reply:
Zastanawiałem się nad tym i nawet zapytałem wujasa czy to zawsze będzie .csv i niby tak, ale w sumie w rzeczywistości to wiadomo jak jest.
IReader będzie lepszy, poprawię.